### ISSUE TITLE
How can we obtain the tokenizer

### ISSUE THREAD

**Issue by ClaireXie** [yaochenzhu/LLM4Rec#5]
Hi Thanks for sharing the code. I wonder how can we obtain the tokenizer. Is it also from GPT-2: https://huggingface.co/openai-community/gpt2/tree/main?

From https://github.com/yaochenzhu/LLM4Rec/blob/main/src/training.py#L145-L148, it looks for a file "vocab_file.json", but I couldn't find it in the GPT-2 repo. Also the tokenizer is sitting under a "tokenizer" folder, which seems not consistent with the one in GPT-2 repo.

Another question, what does "lambda_V" mean and how can I set it?

Thanks!

**Comment by yaochenzhu** [comment]
Hi Claire,

Thanks for reaching out!

I'm sorry for the confusion. vocab_file.json is just the [vocab.json](https://huggingface.co/openai-community/gpt2/blob/main/vocab.json). I'll modify the codes to avoid the confusion.

In addition, for lambda_V, as you can find in our paper, we have two sets of token embeddings for user and items, i.e., collaborative embeddings encoding the interaction information and content embedding encoding the content information. lambda_V denotes the "coupleness" between the user/item collaborative and content embeddings. If lambda_V is large, then the coupleness is tight, where user/item collaborative/content token embeddings are constrained to be more similar (risking introducing unnecessary content information). If lambda_V is small, then the coupleness is loose, where we may risk introducing insufficient content information to support collaborative filtering. Empirically, we find in our paper that lambda_V = 1 serves as a good starting point.

Best,
Yaochen

**Comment by ClaireXie** [comment]
Thanks for the quick reply.

One more question, should I run both training.py and finetuning.py as instructed in the run.sh script?

**Comment by yaochenzhu** [comment]
Yup. They correspond to the pretrain and finetune steps mentioned in the paper.
